document/core/valid/instructions.rst

```diff
@@ -154,6 +154,325 @@ Numeric Instructions
+   \packed(\K{i8x16}) &=& \I32 \\
+   \packed(\K{i16x8}) &=& \I32 \\
+   \packed(t\K{x}N) &=& t \\
+   \end{array}
+
+
+.. _valid-vsunop:
+
+:math:`t\K{.}\vsunop`
+.....................
+
+* The instruction is valid with type :math:`[t] \to [t]`.
+
+.. math::
+   \frac{
+   }{
+     C \vdashinstr t\K{.}\vsunop : [t] \to [t]
+   }
+
+
+.. _valid-vsbinop:
+
+:math:`t\K{.}\vsbinop`
+......................
+
+* The instruction is valid with type :math:`[t~t] \to [t]`.
+
+.. math::
+   \frac{
+   }{
+     C \vdashinstr t\K{.}\vsbinop : [t~t] \to [t]
+   }
+
+
+.. _valid-vsternop:
+
+:math:`t\K{.}\vsternop`
+.......................
+
+* The instruction is valid with type :math:`[t~t~t] \to [t]`.
+
+.. math::
+   \frac{
+   }{
+     C \vdashinstr t\K{.}\vsternop : [t~t~t] \to [t]
+   }
+
+
+.. _valid-vunop:
+
+:math:`vxx\K{.}\vunop`
+......................
+
+* The instruction is valid with type :math:`[\V128] \to [\V128]`.
+
+.. math::
+   \frac{
+   }{
+     C \vdashinstr vxx\K{.}\vunop : [\V128] \to [\V128]
+   }
+
+
+.. _valid-vbinop:
+
+:math:`vxx\K{.}\vbinop`
+.......................
+
+* The instruction is valid with type :math:`[\V128~\V128] \to [\V128]`.
+
+.. math::
+   \frac{
+   }{
+     C \vdashinstr vxx\K{.}\vbinop : [\V128~\V128] \to [\V128]
+   }
+
+
+.. _valid-vternop:
+
+:math:`vxx\K{.}\vternop`
+........................
+
+* The instruction is valid with type :math:`[\V128~\V128~\V128] \to [\V128]`.
+
+.. math::
+   \frac{
+   }{
+     C \vdashinstr vxx\K{.}\vternop : [\V128~\V128~\V128] \to [\V128]
+   }
+
+
+.. _valid-vtestop:
+
+:math:`vxx\K{.}\vtestop`
+........................
+
+* The instruction is valid with type :math:`[\V128] \to [\I32]`.
+
+.. math::
+   \frac{
+   }{
+     C \vdashinstr vxx\K{.}\vtestop : [\V128] \to [\I32]
+   }
+
+.. todo::
+   is this definition of splat okay? Or should we split out i8x16 and i16x8, like below.
+
+
+.. _valid-simd-splat:
+
+:math:`vxx\K{.}\SPLAT`
+..........................
+
+* Let :math:`t` be :math:`\packed(vxx)`.
+* The instruction is valid with type :math:`[t] \to [\V128]`.
+
+.. math::
+   \frac{
+   }{
+     C \vdashinstr vxx\K{.}\SPLAT : [t] \to [\V128]
+   }
+
+
+.. _valid-simd-splat-i8x16:
+
+:math:`\K{i8x16.}\SPLAT`
+..........................
+
+* The instruction is valid with type :math:`[\I32] \to [\V128]`.
+
+.. math::
+   \frac{
+   }{
+     C \vdashinstr \K{i8x16.}\SPLAT : [\I32] \to [\V128]
+   }
+
+
+.. _valid-simd-splat-i16x8:
+
+:math:`\K{i16x8.}\SPLAT`
+..........................
+
+* The instruction is valid with type :math:`[\I32] \to [\V128]`.
+
+.. math::
+   \frac{
+   }{
+     C \vdashinstr \K{i16x8.}\SPLAT : [\I32] \to [\V128]
+   }
+
+
+.. _valid-simd-splat:
+
+:math:`t\K{x}N\K{.}\SPLAT`
+..........................
+
+* The instruction is valid with type :math:`[t] \to [\V128]`.
+
+.. math::
+   \frac{
+   }{
+     C \vdashinstr t\K{x}N\K{.}\SPLAT : [t] \to [\V128]
+   }
+
+
+.. todo::
+   develop some sort of shorthand for this i8x16 and i16x8 repetition?
```

Comment:
Yes, please. :) I think you can just use `\vxx` -- it doesn't matter that this covers additional cases, the grammar definition already narrows the ones that are syntactically allowed.